File: DSA/recursion/factorial.c
#include <stdio.h>
int fact(int n, int product) {
    if(n==1||n==0) {
        return product;
    }
    return fact(n-1, product * n);
}

int main(){
    int n;
    printf("Enter a number:");
    scanf("%d",&n);
    printf("Factorial is %d.\n\n\n\n", fact(n,1));
}

File: Dotnet (C#)/Lab 1/factorial.cs
// WAP to find factorial of a number

using System;

class Factorial {
    static void Main(string[] args) {
        Console.Write("Enter a positive integer: ");
        int n = int.Parse(Console.ReadLine());

        if (n < 0) {
            Console.WriteLine("Factorial is not defined for negative numbers.");
        } else {
            long factorial = 1;
            for (int i = 1; i <= n; i++) {
                factorial *= i;
            }

            Console.WriteLine("Factorial of "+n+" is: "+factorial);
        }
    }
}
